pweave/scripts.py

```diff
@@ -38,6 +47,9 @@ def weave():
     parser.add_option("-t", "--mimetype", dest="mimetype", default=None,
                       help="Source document's text mimetype. This is used to set cell " +
                            "type in Jupyter notebooks")
+    parser.add_option("-v", "--verbose", dest="verbosity",
```

Comment:
I recommend not to use 'v' (nor 'verbose', nor 'verbosity') as name of the feature. It also does not seem to increase verbosity of logging. It seems to propagate logging of ipykernel to stderr.
How about `--show-ipykernel-logs` and 'show_ipykernel_logs'?

Reply:
But I think abukaj's point was that you are increasing verbosity only from one component, not from the whole code base. Hence a more specific name would better communicate what this is doing. On the other hand it is an implementation leak. The user shouldn't have to know what ipykernel is. Not sure what to suggest. It'd be nice to have a true verbose mode that includes but not limited to ipykernel logs. Maybe we could  call this "verbose" for the time being and file an issue towards completing it into a true verbose mode?

Reply:
@piccolbo I concur to have a true verbose mode.
@fgregg For that reason I would use 'verbose' name to control verbosity of the root logger (e.g. to switch it from `WARNING` to `INFO` level), and use other name to include interpreter logs.
How about `--include-interpreter-logs`?

Reply:
This PR also takes a step towards a more complete verbosity by turning the various print statements into logging commands within this code base.

Reply:
@abukaj I'm sorry, but that does not conform with the behavior I would like to have. The use case for this is to understand what went wrong with the python code, and, to me, increasing the verbosity seems like the natural way to achieve that.

Of course, I understand your reasons and preferences, and respect your right to decide. Please let me know if this is a sticking point for you. In that eventuality, I can close this PR.

Reply:
@fgregg to be clear: I like the new features, my point regards the interface.
I stick that `-v` and `--verbose` should control verbosity of the root logger (we can also provide even `--debug`, `-d` switch for `DEBUG` level), but it is @mpastell who is the final authority.

Reply:
I understand. Thank you for you kind attention, and good luck.

Reply:
From a UI perspective, I don't think we should have separate verbose modes for different subsystems. The fact that there are two loggers in the implementation, there may be a good reason for that which I don't know, but passing that onto the user, I don't think that's a good idea. What does the user know about interpreters, loggers? They just had a pweave failure and they want to know what's going on. -v and take a look. Then they see something like

chunk number (from root logger)
python errors (from kernel logger)
chunk number
python errors
...

What is wrong with that?
We can always add more detail later, but removing options is hard (others say you should never do it).

Reply:
@piccolbo then we should include ipykernel logs in output all the time.

Once again it is about intuitiveness (in this time of the UI). When I use `pweave --verbose` I expect to see `Processing chunk ...` messages (which I prefer to be hidden by default).
It is what `-v` does in almost every tool I can think about.
Moreover, such feature may be later easily extended with `-d`, `--debug`.

Without @mpastell I can accept the implemented feature with (possibly temporary) any other switch not conflicting with some obvious feature which may be implemented.
It would be easier to add more switches supporting that feature later, than to remove them -- or even worse, to change their meaning.